Add more handbook-first guidance (#924)
sourcegraph/handbook: content/handbook/index.md

- **The Handbook can ve viewed by anyone.** One of our Sourcegraph values is transparency. We want all public information to be available to everyone, whether they work at Sourcegraph or not, and it all lives in the Handbook. The Handbook should be searchable and easy to navigate.
- **The Handbook can be edited by any Sourcegraph teammate.** We value high agency and encourage all teammates to update the Handbook as they see fit. We provide resources to help teammates contribute to the Handbook no matter their technical background.
- **The Handbook is maintained by every Sourcegraph teammate.** At Sourcegraph, we work as a team. No one person is responsible for keeping our content updated, it falls on all of us. Editing and updating pages should be quick and well documented.

> - **The Handbook is a source of truth** at Sourcegraph, and information there is expected to be accurate and up-to-date. This includes for different processes the company uses, for something to be in force as a practice here at Sourcegraph, it should be in the handbook. A Google Doc or pending PR to the handbook is not sufficient as a source of truth because using those reduces the effectiveness of the handbook as a trusted source of knowledge. Further, by never needing to merge your change to bring it into force, it discourages making small, iterative improvements.

Reviewer: ```suggestion
- **The Handbook is a source of truth** at Sourcegraph, and its information is expected to be accurate and up-to-date. It should be documented in the handbook for any company process to be in force as a practice here at Sourcegraph. A Google Doc or pending PR to the handbook is not sufficient as a source of truth because using those reduces the effectiveness of the handbook as a trusted source of knowledge. Further, never needing to merge your change to bring it into force discourages making small, iterative improvements.
```

> - **Use PRs to discuss changes to the handbook.** Whenever possible (an exception, for example, might be when discussing non-public information that might lead to a handbook update), use PRs to propose changes to the handbook. Using a Google Doc instead can be easy to miss existing (even conflicting) content. Getting to a PR more quickly makes it easier to discover these and for others to contribute cross-links or highlight duplicated/conflicting content.

Reviewer 2: @jyavorska this feels like it belongs in "Handbook Usage" rather than "guiding principles" to me. Both of these additions read more like best practices than a Handbook principle.

Author: Sure that's fine, moved both down @marybelzer

Author: Moved

> - **Use PRs to discuss changes to the handbook.** Whenever possible (an exception for example might be when discussing non-public information that might lead to a handbook update), use PRs to the handbook to propose changes to the handbook. When using a Google Doc instead it can be easy to miss that there's existing (even conflicting) content, and by getting to a PR more quickly, it's easier to discover these and for others to contribute cross-links or highlight duplicated/conflicting content.

Reviewer: ```suggestion
- **Use PRs to discuss changes to the handbook.** Whenever possible (an exception, for example, might be when discussing non-public information that might lead to a handbook update), use PRs to propose changes to the handbook. Using a Google Doc instead can be easy to miss existing (even conflicting) content. Getting to a PR more quickly makes it easier to discover these and for others to contribute cross-links or highlight duplicated/conflicting content.
```

## Handbook Usage

- The Handbook is a source of truth, and its information is expected to be accurate and up-to-date. For any process to be in force, it should be documented in the handbook. A Google Doc or pending PR to the handbook is not sufficient as a source of truth because using those reduces the effectiveness of the handbook as a trusted source of knowledge. Further, never needing to merge your change to bring it into force discourages making small, iterative improvements.

> - Use PRs to discuss changes to the handbook. Whenever possible (an exception, for example, might be when discussing non-public information that might lead to a handbook update), use PRs to propose changes to the handbook. Using a Google Doc instead can be easy to miss existing (even conflicting) content. Getting to a PR more quickly makes it easier to discover these and for others to contribute cross-links or highlight duplicated/conflicting content.

Reviewer 2: @jyavorska Looking at this a bit closer, it's a little repetitive of what's in lines 53-55, but does add some good additional detail. Possible to consolidate?

Author: @marybelzer it looks like you were commenting on an outdated version so I'm not following your suggestion. Can you check again? Or perhaps go ahead and make the change you want?
